lib/archethic_web/live/last_ten_transactions.ex

```diff
@@ -0,0 +1,157 @@
+defmodule ArchethicWeb.LastTenTransactionsComponent do
+  @moduledoc """
+    Live component for Dashboard Explorer to display last 10 transactions
+  """
+
+  use ArchethicWeb, :live_component
+
+  alias Archethic.BeaconChain
+
+  alias Archethic.Election
+
+  alias Archethic.P2P
+  alias Archethic.P2P.Node
+  alias Archethic.P2P.Message.GetCurrentSummaries
+  alias Archethic.P2P.Message.TransactionSummaryList
+
+  alias ArchethicWeb.ExplorerLive.LastTenTransactionCache
+
+  def mount(socket) do
+    socket =
+      socket
+      |> assign(:transactions, [])
+
+    {:ok, socket}
+  end
+
+  def update(
+        %{transaction: transaction} = _assigns,
+        socket
+      )
+      when not is_nil(transaction) do
+    {:ok, transactions} =
+      LastTenTransactionCache.resolve_put(transaction, fn ->
+        fetch_last_ten_transactions()
+      end)
+
+    socket =
+      socket
+      |> assign(:transactions, transactions)
+
+    {:ok, socket}
+  end
+
+  def update(assigns, socket) do
+    {:ok, transactions} =
+      LastTenTransactionCache.resolve(fn ->
+        fetch_last_ten_transactions()
+      end)
+
+    socket = socket |> assign(assigns) |> assign(transactions: transactions)
+    {:ok, socket}
+  end
+
+  def render(assigns) do
+    ~L"""
+      <div class="box mb-2">
+        <p>Latest Txns</p>
+      <div>
+      <div>
+        <%= for tx <- @transactions do %>
+          <div class="columns">
+            <div class="column is-7-desktop">
+              <%= link to: Routes.live_path(@socket, ArchethicWeb.TransactionDetailsLive, Base.encode16(tx.address)) do%>
+                <span class="text_wrap has-text-primary is-size-6"><%= Base.encode16(tx.address) %></span>
+                <% end %>
+            </div>
+            <div class="column is-2-desktop is-size-6">
+              <%= format_date(tx.timestamp) %>
+            </div>
+            <div class="column is-2-desktop is-size-6">
+              <span class="tag is-light is-info"><%= tx.type %></span>
+            </div>
+          </div>
+        <% end %>
+      </div>
+    """
+  end
+
+  defp fetch_previous_dates(time = %DateTime{}, txns) when length(txns) < 10 do
+    previous_time = BeaconChain.previous_summary_time(time)
+    txns = txns ++ list_transactions_from_summary(previous_time)
+    fetch_previous_dates(previous_time, txns)
+  end
+
+  defp fetch_previous_dates(_time = %DateTime{}, txns) when length(txns) >= 10 do
+    txns
+  end
+
+  defp fetch_last_ten_transactions() do
```

Comment:
the same here about naming.
the module or the function could define arguments to determine the size of list instead.

Reply:
Yes that would be better.